query/src/sessions/session_mgr.rs

```diff
@@ -201,7 +201,8 @@ impl SessionManager {
             &self.conf.query.cluster_id,
         );
 
-        self.active_sessions.write().remove(session_id);
+        let mut sessions = futures::executor::block_on(self.active_sessions.write());
```

Comment:
await?

Reply:
This function is eventually called by drop function that impl Drop trait, async drop still in rust roadmap...

Reply:
Can you make `active_sessions` an `std::sync::Mutex`?
Then the `await` can be get rid of.

But using a sync-mode mutex introduces some retry-loops when inserting items into it.
Because it can not be held across an `await`.

Reply:
i guess std::sync::Mutex is ok in this case, but should be careful NOT to introduce any async/await point inside the mutex guard.

1. sync mutex lock
2. await yield out to another coroutine
3. another coroutine tries to acquire the lock

however, it's a good practise to not do any blocking operations inside a mutex guard